python/ray/util/sgd/torch/torch_trainer.py

```diff
@@ -678,6 +715,7 @@ def step(self):
         train_stats = self.trainer.train(max_retries=10, profile=True)
         validation_stats = self.trainer.validate(profile=True)
         stats = merge_dicts(train_stats, validation_stats)
+        self._iter += 1
```

Comment:
Is this state supposed to be saved upon checkpoint?

Reply:
Ah yes good catch. Added it to save_checkpoint and load_checkpoint.

Reply:
Actually, now I know what you're doing here, I think we already provide `self.training_iteration` in trainable, so you don't actually need to save/manage this stuff.

Reply:
Ah got it. Changed it to using self.training_iteration and removed the duplicate metadata checkpointing.